lib/rack/attack/throttle.rb

```diff
@@ -26,12 +26,15 @@ def [](req)
         current_limit  = limit.respond_to?(:call) ? limit.call(req) : limit
         key            = "#{name}:#{discriminator}"
         count          = cache.count(key, current_period)
+        epoch_time     = cache.last_epoch_time
```

Comment:
I wonder whether making the epoch time an argument to the `count` method might end up feeling a bit cleaner... Instead of keeping the last epoch time in the cache.

Not a blocker for merging though, just a consideration for a possible future refactor :-)